mautic/mautic
# M3: Change UTF8 to utf8mb4 for new installation

**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | 
| New feature? | 
| Automated tests included? |
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | 
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:

This PR is follow up of this PR https://github.com/mautic/mautic/pull/6554

Based on this discussion https://github.com/mautic/mautic/issues/6490 and this article https://mathiasbynens.be/notes/mysql-utf8mb4
This PR add as default charset  utf8mb4  which allow use all unicode symbols (like emoji)

Conclusion from article 

> Never use utf8 in MySQL — always use utf8mb4 instead. Updating your databases and code might take some time, but it’s definitely worth the effort. Why would you arbitrarily limit the set of symbols that can be used in your database? Why would you lose data every time a user enters an astral symbol as part of a comment or message or whatever it is you store in your database? There’s no reason not to strive for full Unicode support everywhere. Do the right thing, and use utf8mb4. 🍻


#### Steps to test this PR:
1.  Create new installation
2. Go to emails, create email with 🥐 emoji  in subject 
3. Should works properly.
4. The old installation with this code applied should work as before.

#### List backwards compatibility breaks:
Hard to say. We have to setup VARCHAR to 191 lenght. More in https://github.com/FriendsOfSymfony/FOSUserBundle/issues/1919#issuecomment-165066886
